File: ViewBinding/viewBindingFragDemo/app/src/main/java/edu/cs4730/viewbindingfragdemo/OneFragment.java
package edu.cs4730.viewbindingfragdemo;

import android.app.Activity;
import android.content.Context;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import edu.cs4730.viewbindingfragdemo.databinding.FragmentOneBinding;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

/**
 *  simple fragment has have a edittext and button.  using the viewbinding so you
 *  don't need findviewbyid.  and you know everything is not null, plus this
 *  also has an advantage you have the same id names in different xml and not have an issues
 *  with most studio picking the wrong one.
 */
public class OneFragment extends Fragment {

    private FragmentOneBinding binding;
    private OnFragmentInteractionListener1 mListener;

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        binding = FragmentOneBinding.inflate(inflater, container, false);

        binding.btnSubmit.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if (mListener != null) {
                    String value = "stuff";
                    if (binding.etName.getText().toString().compareTo("") != 0)
                        value = binding.etName.getText().toString();
                    mListener.onFragmentInteraction1(value);
                }
            }
        });

        //needs to return a view.  so get it for the return
        return binding.getRoot();
    }


    @Override
    public void onDestroyView() {
        super.onDestroyView();
        binding = null;
    }

    @Override
    public void onAttach(@NonNull Context context) {
        super.onAttach(context);
        Activity activity = requireActivity();
        try {
            mListener = (OnFragmentInteractionListener1) activity;
        } catch (ClassCastException e) {
            throw new ClassCastException(activity.toString()
                + " must implement OnFragmentInteractionListener");
        }
    }

    @Override
    public void onDetach() {
        super.onDetach();
        mListener = null;
    }

    public interface OnFragmentInteractionListener1 {
        void onFragmentInteraction1(String Data);
    }
}

File: ViewBinding/viewBindingFragDemo_kt/app/src/main/java/edu/cs4730/viewbindingfragdemo_kt/OneFragment.kt
package edu.cs4730.viewbindingfragdemo_kt

import android.app.Activity
import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import edu.cs4730.viewbindingfragdemo_kt.databinding.FragmentOneBinding

/**
 * simple fragment has have a edittext and button.  using the viewbinding so you
 * don't need findviewbyid.  and you know everything is not null, plus this
 * also has an advantage you have the same id names in different xml and not have an issues
 * with most studio picking the wrong one.
 */
class OneFragment : Fragment() {
    private lateinit var binding: FragmentOneBinding
    private var mListener: OnFragmentInteractionListener1? = null

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        binding = FragmentOneBinding.inflate(inflater, container, false)
        binding.btnSubmit.setOnClickListener {
            if (mListener != null) {
                var value = "stuff"
                if (binding.etName.text.toString().compareTo("") != 0)
                    value = binding.etName.text.toString()
                mListener!!.onFragmentInteraction1(value)
            }
        }

        //needs to return a view.  so get it for the return
        return binding.root
    }

    override fun onAttach(context: Context) {
        super.onAttach(context)
        val activity: Activity = requireActivity()
        mListener = try {
            activity as OnFragmentInteractionListener1
        } catch (e: ClassCastException) {
            throw ClassCastException(
                activity.toString()
                        + " must implement OnFragmentInteractionListener"
            )
        }
    }

    override fun onDetach() {
        super.onDetach()
        mListener = null
    }

    interface OnFragmentInteractionListener1 {
        fun onFragmentInteraction1(Data: String?)
    }
}